crates/oxc_ast/src/ast_impl/js.rs

```diff
@@ -1306,28 +1381,32 @@ impl PropertyDefinitionType {
 }
 
 impl MethodDefinitionKind {
+    #[inline]
     pub fn is_constructor(&self) -> bool {
         matches!(self, Self::Constructor)
     }
 
+    #[inline]
     pub fn is_method(&self) -> bool {
         matches!(self, Self::Method)
     }
 
+    #[inline]
     pub fn is_set(&self) -> bool {
         matches!(self, Self::Set)
     }
 
+    #[inline]
     pub fn is_get(&self) -> bool {
         matches!(self, Self::Get)
     }
 
-    pub fn scope_flags(self) -> ScopeFlags {
+    pub const fn scope_flags(self) -> ScopeFlags {
         match self {
-            Self::Constructor => ScopeFlags::Constructor | ScopeFlags::Function,
+            Self::Constructor => ScopeFlags::Constructor.union(ScopeFlags::Function),
```

Comment:
1. As with above comments, this should be a separate PR.
2. Personally I prefer it as it was before. I find ` | ` clearer than `.union()`.
3. What is the rationale for making this function `const`?